commentwriter.go

```diff
@@ -0,0 +1,42 @@
+package vfsgen
+
+import "io"
+
+// commentWriter writes given input to underlying io.Writer as a Go comment,
+// using line comments (//).
+type commentWriter struct {
+	io.Writer
+
+	wroteComment bool
+}
+
+func (cw *commentWriter) Write(p []byte) (n int, err error) {
+	for i, c := range p {
+		if c == '\n' {
+			if !cw.wroteComment {
+				_, err := cw.Writer.Write([]byte("//"))
+				if err != nil {
+					return n, err
+				}
+			}
+			cw.wroteComment = false
+		} else {
+			if !cw.wroteComment {
+				_, err := cw.Writer.Write([]byte("// "))
+				if err != nil {
+					return n, err
+				}
+				cw.wroteComment = true
+			}
+		}
+		_, err := cw.Writer.Write(p[i : i+1])
+		if err != nil {
+			return n, err
+		}
+		n++
+	}
+	if n != len(p) {
+		return n, io.ErrShortWrite
+	}
```

Comment:
You're good. You will always hit the increment unless you error out first.  You might consider using a compound if-clause and no short variable declaration to avoid allocating and shadowing `err` unnecessarily, since you otherwise don't appear to be using it.

Since the final check on n is unnecessary, you might consider incrementing it by 2 when you write an empty comment and 3 when you write the beginning of a comment line to reflect the actual bytes you've written, in case this is useful to the caller in debugging.